Core/Core/Credentials/AccountManager.cs

```diff
@@ -195,6 +223,25 @@ public static string GetDefaultServerUrl()
     return serverUrl;
   }
 
+  /// <summary>
+  /// Upgrades an account from https://speckle.xyz to https://app.speckle.systems
+  /// </summary>
+  /// <param name="id"></param>
+  public static void UpgradeAccount(string id)
+  {
+    var account =
+      GetAccounts().FirstOrDefault(acc => acc.id == id)
+      ?? throw new SpeckleAccountManagerException($"Account {id} not found");
+
+    if (!account.serverInfo.url.Contains("https://speckle.xyz"))
+    {
+      throw new SpeckleAccountManagerException($"Can only upgrade accounts on speckle.xyz");
+    }
+
+    account.serverInfo.url = DEFAULT_SERVER_URL;
+    s_accountStorage.UpdateObject(account.id, JsonConvert.SerializeObject(account));
+  }
+
```

Comment:
This implementation is **not correct** ⚠️.

We should be using `migration.migratedTo` to determine if an account "can be upgraded" and if so, use said url as the one to swap.

Also, this is out of scope for this ticket, or did I miss something in the past days while i was lost in Digicert? 🙂

Reply:
I think this might just be matteo original implementation, Connors pr is still draft, so I imagine there's more to come.